samtranslator/plugins/api/implicit_api_plugin.py

```diff
@@ -93,6 +93,19 @@ def _generate_implicit_api_resource(self) -> Dict[str, Any]:
         Helper function implemented by child classes that create a new implicit API resource
         """
 
+    def _add_tags_to_implicit_api_if_necessary(self, resource: SamResource, template: SamTemplate) -> None:
+        """
+        Decides whether to add tags to the implicit api resource.
+        :param dict template_dict: SAM template dictionary
+        """
+        implicit_api_resource = template.get(self.IMPLICIT_API_LOGICAL_ID)
+        globals_var = template.get_globals().get(SamResourceType(resource.type).name, {})
+        should_propagate_tags = resource.properties.get("PropagateTags") or globals_var.get("PropagateTags")
+
+        if implicit_api_resource and resource.properties.get("Tags") and should_propagate_tags:
+            implicit_api_resource.properties.setdefault("Tags", {}).update(resource.properties["Tags"])
```

Comment:
I don't really like how there's so many assumptions and mixing of concerns. It takes a generic resource as input, yet assumes the tags is `Tags`, the propagate tags is `PropagateTags`, but there's nothing guaranteeing it's correct from this function's perspective.

Reply:
This `resource` is a SAM resource not a generic CFN resource. We can safely assume that all SAM resources have use `Tags` property isntead of an arbitrary name.